client/lib/sites-list/list.js

```diff
@@ -228,6 +228,10 @@ SitesList.prototype.update = function( sites ) {
 				return siteObj;
 			}
 
+			if ( siteObj.options.page_on_front !== site.options.page_on_front ) {
```

Comment:
A comment as to why this is being done would be helpful, as without the GH PR description, I wouldn't immediately understand why this was being done.